vumi/persist/tests/test_fake_redis.py

```diff
@@ -178,6 +186,14 @@ def test_zremrangebyrank_negative_stop(self):
             [('one', 1)], 'zrange', 'set', 0, -1, withscores=True)
 
     @inlineCallbacks
+    def test_zremrangebyrank_negative_stop_empty_range(self):
+        yield self.redis.zadd('set', one=1, two=2, three=3)
+        yield self.assert_redis_op(0, 'zremrangebyrank', 'set', 0, -5)
```

Comment:
I was thinking of a case where we do something like:

```
ZREMRANGEBYRANK set -5 -4
```

(where `set` has a length of `3`).

With [the current way](https://github.com/praekelt/vumi/blob/develop/vumi/persist/fake_redis.py#L506-L507) we work with negative start values, I think we would normalize `-5` to `-2` and `-4` to `-1`, so we would end up removing a non-empty range of values. I thought it might be useful to add a test that fails for the current implementation and passes for the fix in this PR.

I think the current tests are sufficient for checking how we handle both empty and non-empty ranges when we use negative start and end values, but just thought it might be worth adding a test for something that is currently failing. Not very important though, your call really.